lib/FreeRTOS/mqtt-agent/mqtt_agent.c

```diff
@@ -457,145 +464,81 @@ static MQTTStatus_t addCommandToQueue( AgentMessageContext_t * pQueue,
-
-                /* Add to pending ack list, or call callback if QoS 0. */
-                addAckToList = ( pPublishInfo->qos != MQTTQoS0 ) && ( operationStatus == MQTTSuccess );
-
-                break;
-
-            case SUBSCRIBE:
-            case UNSUBSCRIBE:
-                pSubscribeArgs = ( MQTTAgentSubscribeArgs_t * ) ( pCommand->pArgs );
-                packetId = MQTT_GetPacketId( pMQTTContext );
-
-                if( pCommand->commandType == SUBSCRIBE )
-                {
-                    /* Even if some subscriptions already exist in the subscription list,
-                     * it is fine to send another subscription request. A valid use case
-                     * for this is changing the maximum QoS of the subscription. */
-                    operationStatus = MQTT_Subscribe( pMQTTContext,
-                                                      pSubscribeArgs->pSubscribeInfo,
-                                                      pSubscribeArgs->numSubscriptions,
-                                                      packetId );
-                }
-                else
-                {
-                    operationStatus = MQTT_Unsubscribe( pMQTTContext,
-                                                        pSubscribeArgs->pSubscribeInfo,
-                                                        pSubscribeArgs->numSubscriptions,
-                                                        packetId );
-                }
-
-                addAckToList = ( operationStatus == MQTTSuccess );
-                break;
-
-            case PING:
-                operationStatus = MQTT_Ping( pMQTTContext );
-
-                break;
-
-            case CONNECT:
-                pConnectArgs = ( MQTTAgentConnectArgs_t * ) ( pCommand->pArgs );
-                operationStatus = MQTT_Connect( pMQTTContext,
-                                                pConnectArgs->pConnectInfo,
-                                                pConnectArgs->pWillInfo,
-                                                pConnectArgs->timeoutMs,
-                                                &( pConnectArgs->sessionPresent ) );
-                break;
-
-            case DISCONNECT:
-                operationStatus = MQTT_Disconnect( pMQTTContext );
-                runProcessLoops = false;
-
-                break;
+        assert( pCommand->commandType < NUM_COMMANDS );
+        commandFunction = pCommandFunctionTable[ pCommand->commandType ];
+        pCommandArgs = pCommand->pArgs;
+    }
+    else
+    {
+        commandFunction = pCommandFunctionTable[ NONE ];
+    }
 
-            case TERMINATE:
-                LogInfo( ( "Terminating command loop.\n" ) );
-                runProcessLoops = false;
+    operationStatus = commandFunction( pMqttAgentContext, pCommandArgs, &commandOutParams );
 
-            default:
-                break;
-        }
+    if( commandOutParams.addAcknowledgment )
```

Comment:
```suggestion
    if( commandOutParams.addAcknowledgment == true )
```

Reply:
I don't really agree with this. I'll change it if MISRA asks, but since we now include stdbool.h it shouldn't complain any more.

Reply:
If we are including stdbool.h, that should be fine, but then my question would be about the consistency. We now have 2 styles in the code. Can we use one style in the code then? May be that can be a separate PR.

Reply:
I've consistently been dropping the `== true` in this code and we have more instances where we don't use it, so I'd prefer to leave this as is